runtime/metricsview/ast.go

```diff
@@ -623,6 +628,70 @@ func (a *AST) buildUnderlyingWhere() (*ExprNode, error) {
 
 // buildBaseSelect constructs a base SELECT node against the underlying table.
 func (a *AST) buildBaseSelect(alias string, comparison bool) (*SelectNode, error) {
+	if a.query.inlineBaseSelect && !comparison {
+		n := &SelectNode{
+			Alias:     alias,
+			DimFields: a.dimFields,
+			Group:     true,
+			FromTable: a.underlyingTable,
+			IsInline:  true,
+		} // keep the dims and from table so that while adding time comparison measures to the AST structure is maintained
+
+		// Add the dimension fields to the inline view.
+		inDims := make([][]FieldNode, 0, len(a.query.inlineDims))
+		for dim, values := range a.query.inlineDims {
+			fields := make([]FieldNode, 0, len(values))
+			for _, v := range values {
+				f := FieldNode{
+					Name: dim.(string),
+				}
+				switch v := v.(type) {
+				case time.Time:
+					if a.dialect == drivers.DialectClickHouse {
+						f.Expr = fmt.Sprintf("parseDateTimeBestEffort('%s')", v.Format(time.RFC3339))
+					} else {
+						f.Expr = fmt.Sprintf("CAST('%s' AS TIMESTAMP)", v.Format(time.RFC3339))
+					}
+				case bool:
+					f.Expr = fmt.Sprintf("%v", v)
+				default:
+					f.Expr = fmt.Sprintf("'%v'", v)
+				}
+				if v == nil {
+					f.Expr = nilExpr
+				}
+				fields = append(fields, f)
+			}
+			inDims = append(inDims, fields)
+		}
+
+		// Add the measure fields to the inline view.
+		inMeasures := make([][]FieldNode, 0, len(a.query.inlineMeasures))
+		for measure, values := range a.query.inlineMeasures {
+			fields := make([]FieldNode, 0, len(values))
+			for _, v := range values {
+				if v == nil {
+					v = nilExpr
```

Comment:
Why can't we use `null` in SQL to represent nils?

Reply:
outdated and removed, however we are using NULL while writing sql when we encounter this exception